backend/ml/mehestan/primitives.py

```diff
@@ -30,7 +31,15 @@ def QrMed(W: float, w: Union[pd.Series, float], x: pd.Series, delta: pd.Series):
 
     def L_prime(m: float):
         x_minus_m = x - m
-        return W * m - np.sum(w * x_minus_m / np.sqrt(delta_2 + x_minus_m ** 2))
+        denominator = np.sqrt(delta_2 + x_minus_m ** 2)
+        # Avoid division by 0 by setting the result of the division
+        # to the sign of x_minus_m (+1 or -1) if the denominator is 0
+        division_result = np.where(
+            denominator > 0,
+            np.divide(x_minus_m, denominator, where=denominator > 0),
+            np.sign(x_minus_m),
```

Comment:
`denominator` is 0 only when both `delta` and `x_minus_m` are 0. So `np.sign(x_minus_m)` can be replaced with 0.

Note that `L_prime` is on the critical path from a performance point of view. So it would be nice to move the complexity outside of this function as much as possible.  
Another approach could be to define `delta_2`  so that it can no longer be zero: 
```py
delta_2 = np.where(delta > 0, delta ** 2, np.spacing(0))
```